mod/01_rating/set/type/metric_value.rb

```diff
@@ -334,7 +332,24 @@ def special_editor? args
     end
   end
 
-  view :metric_company_add_value_editor do |_args|
+  def find_potential_sources company, metric
+    Card.search type_id: Card::SourceID,
+                right_plus: ['company', { refer_to: company }],
+                and: {
+                  right_plus: [
+                    'report_type',
+                    refer_to: {
+                      referred_to_by: metric + '+report_type' }] }
+  end
+
+  view :metric_company_add_value_editor do |args|
+    sources = find_potential_sources args[:company], args[:metric]
+    relevant_sources =
+      if sources.empty?
+        'None'
+      else
+        sources.map(&:name).join(',')
+      end
```

Comment:
fwiw, this could be a one-liner inside the interpreted HAML.

`sources.empty? ? 'None' : sources.map(&:name) * ','`